File: 2018/day02/solution.rb
def cs (id)
    counts = Array.new(26, 0)
    id.chars.map(&:ord).each do |i|
        counts[i-97] += 1
    end
    return counts.include?(2), counts.include?(3)
end

input = File.open("input.txt").read.split"\n"

twos,threes = 0,0
input.each do |line|
    line.delete! "\n"
    x, y = cs line
    if x then twos+=1 end
    if y then threes+=1 end
end

puts twos * threes

input.each_with_index do |line, idx|
    input[idx, input.length - idx].each do |other|
        count = 0
        line.length.times do |i|
            if line[i] == other[i] then count += 1 end
        end
        if count == 1 
            then 
            puts [line, other]
            break
        end
    end
end

File: 2018/day02/solution.py
text = open("input.txt", "r").read().split("\n")


def cs(id):
    counts = [0] * 26
    for c in id:
        counts[ord(c) - ord("a")] += 1
    return any(map(lambda x: x == 2, counts)), any(map(lambda x: x == 3, counts))

twos = 0
threes = 0
for line in text:
    x,y = cs(line)
    twos += x
    threes += y

print(twos * threes)

def comp(l, o):
    count = 0
    for i in range(len(l)):
        if not l[i] == o[i]:
            count += 1
    return count


for i in range(len(text)):
    l = text[i]
    for o in text[i+1:]:
        if comp(l,o) == 1:
            print(l,o)
